aws-xray-recorder-sdk-sql/src/main/java/com/amazonaws/xray/sql/TracingStatement.java

```diff
@@ -0,0 +1,134 @@
+        return (Statement) Proxy.newProxyInstance(TracingConnection.class.getClassLoader(),
+                new Class[] { resolveProxiedInterface(invokedMethodName) },
+                new TracingStatementHandler(s, resolveSqlQuery(invokedMethodName, invokedMethodArgs)));
+    }
+
+    private static Class resolveProxiedInterface(String invokedMethodName) {
+        switch (invokedMethodName) {
+            case "prepareCall": return CallableStatement.class;
+            case "prepareStatement": return PreparedStatement.class;
+            case "createStatement": return Statement.class;
+            default: throw new IllegalStateException("TracingSegment can not decorate " + invokedMethodName);
+        }
+    }
+
+    private static String resolveSqlQuery(String invokedMethodName, Object[] invokedMethodArgs) {
+        switch (invokedMethodName) {
+            case "prepareCall":
+            case "prepareStatement":
+                return (String) invokedMethodArgs[0];
+            case "createStatement":
+                return null; //the correct sql will be retrieved later in resolveExecutedSql
+            default: throw new IllegalStateException("TracingSegment can not resolve sql for " + invokedMethodName);
+        }
+    }
+
+    private static class TracingStatementHandler implements InvocationHandler {
+
+        private final Statement original;
+        private final String sql;
+
+        TracingStatementHandler(Statement original, String sql) {
+            this.original = original;
+            this.sql = sql;
+        }
+
+        public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+            //mostly based on https://github.com/aws/aws-xray-sdk-java/blob/master/aws-xray-recorder-sdk-sql-mysql/src/main/java/com/amazonaws/xray/sql/mysql/TracingInterceptor.java
+            // but in a more straightforward logic (IMHO :) )
+            if (method.getName().equals("execute") || method.getName().equals("executeQuery") || method.getName().equals("executeUpdate") || method.getName().equals("executeBatch")) {
+                //invoke the original method "wrapped" in a XRay Subsegment
+                //implementation note : can not use shorter AWSXRay.createSubsegment because we need to deal with InvocationTargetException
+                boolean openDummySegment = false;
+                if (! AWSXRay.getCurrentSegmentOptional().isPresent()) {
+                    //we are (probably) being executed outside AWS env (unit tests, background process, ...)
+                    // -> we do not want to crash or to log a SegmentNotFoundException, so we print to stdout
+                    // note that if you are in AWS context, stdout will be log to cloudwatch logs anyway
+                    openDummySegment = true;
+                    System.out.println(sanitizeSql(resolveExecutedSql(method.getName(), args)));
```

Comment:
We generally don't want to print to standard out, as this could bloat the customer's application. Logging is better because customers have log level control.